scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java

```diff
@@ -812,11 +814,32 @@ private void recover(RecoveredSchedulerState recoveredState) {
         }
     }
 
-    private void inErrorTasksRecovered(Vector<InternalJob> runningJobs) {
-        Iterator<InternalJob> iterJob = runningJobs.iterator();
+    private void recoverTasksState(Vector<InternalJob> jobs, boolean restoreInErrorTasks) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Replace the synchronized class "Vector" by an unsynchronized one such as "ArrayList" or "LinkedList". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1149)